tests/fsharp/single-test.fs

```diff
@@ -161,136 +150,101 @@ let generateProjectArtifacts (pc:ProjectConfiguration) targetFramework configura
+
+    projBody
+
+let potentialSources = ["testlib.fsi";"testlib.fs";"test.mli";"test.ml";"test.fsi";"test.fs";"test2.fsi";"test2.fs";"test.fsx";"test2.fsx"]
+
+let executeSingleTestBuildAndRun cfg outputType testCompilerVersion targetFramework optimize extraRef =
     let loadSources = []
     let useSources = []
-    let extraSources = ["testlib.fsi";"testlib.fs";"test.mli";"test.ml";"test.fsi";"test.fs";"test2.fsi";"test2.fs";"test.fsx";"test2.fsx"]
+    let extraSources = potentialSources |> List.filter (fileExists cfg)
     let utilitySources = [__SOURCE_DIRECTORY__  ++ "coreclr_utilities.fs"]
-    let referenceItems =  if String.IsNullOrEmpty(copyFiles) then [] else [copyFiles]
-    let framework = "netcoreapp2.0"
-
-    let executeSingleTestBuildAndRun outputType compilerType targetFramework optimize =
-        let mutable result = false
-        let directory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName() )
-        let pc = {
-            OutputType = outputType
-            Framework = framework
-            SourceDirectory = cfg.Directory
-            SourceItems = sources
-            ExtraSourceItems = extraSources
-            UtilitySourceItems = utilitySources
-            ReferenceItems = referenceItems
-            LoadSources = loadSources
-            UseSources = useSources
-            Optimize = optimize
-        }
-
-        let targetsBody = generateTargets
-        let overridesBody = generateOverrides
-        let targetsFileName = Path.Combine(directory, "Directory.Build.targets")
-        let propsFileName = Path.Combine(directory, "Directory.Build.props")
-        let overridesFileName = Path.Combine(directory, "Directory.Overrides.targets")
-        let projectFileName = Path.Combine(directory, Path.GetRandomFileName() + ".fsproj")
-        try
-            // Clean up directory
-            Directory.CreateDirectory(directory) |> ignore
-            copyFilesToDest cfg.Directory directory
-            try File.Delete(Path.Combine(directory, "FSharp.Core.dll")) with _ -> ()
-            emitFile targetsFileName targetsBody
-            emitFile overridesFileName overridesBody
-            if outputType = OutputType.Exe then
-                let executeFsc testCompilerVersion targetFramework =
-                    let propsBody = generateProps testCompilerVersion cfg.BUILD_CONFIG
-                    emitFile propsFileName propsBody
-                    let projectBody = generateProjectArtifacts pc targetFramework cfg.BUILD_CONFIG
-                    emitFile projectFileName projectBody
-                    use testOkFile = new FileGuard(Path.Combine(directory, "test.ok"))
-                    exec { cfg with Directory = directory }  cfg.DotNetExe (sprintf "run -f %s" targetFramework)
-                    testOkFile.CheckExists()
-                executeFsc compilerType targetFramework
```

Comment:
@KevinRansom I notice `result <- true` wasn't being set for the `fsc` path which was a little odd

Reply:
I suppose it just meant that there was no exception, so the "printing" in the finally block was being executed